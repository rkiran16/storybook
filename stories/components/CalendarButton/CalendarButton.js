import { html } from "lit-html";

export const CalendarButton = ({checked, disabled, isCurrentDay}) => {
   const todayHtml = isCurrentDay ? html`<span class="material-icons today-indicator text-primary">fiber_manual_record</span>` : '' ;
   const currentDayClass = isCurrentDay ? 'current-day': '';
   const calendarClass =  `calendar-btn-wrapper ${currentDayClass}`;

   return html`
    <div class=${calendarClass}>
      ${todayHtml}
      <input type="checkbox" class="btn-check" id="btn-check" ?disabled=${disabled} ?checked=${checked}  autocomplete="off">
      <label class="btn" for="btn-check">
        <div class="d-flex flex-column justify-content-evenly">
          <span class="material-icons icon">done</span>
          <span class="day">Sat</span>
          <span class="date">5</span>
        </div>
      </label>
     </div>
   `;
}