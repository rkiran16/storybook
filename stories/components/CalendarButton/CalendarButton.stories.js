import { CalendarButton } from "./CalendarButton";

export default {
   title: 'Components/Buttons/CalendarButton',
   argTypes: {
      checked: {
         name: "Selected",
         type: { name: 'boolean', required: false},
         description: "Selected state of Calendar button",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: {type: 'boolean'}
      },
      disabled: {
         name: "Disabled",
         type: { name: 'boolean', required: false},
         description: "Use when the user cannot proceed or until an input is collected",
         defaultValue: false,
         table: {
           type: { summary: 'boolean' },
           defaultValue: { summary: false },
         },
         control: {
           type: 'boolean'
         }
      },
      isCurrentDay: {
         name: "Current Day",
         type: { name: 'boolean', required: false},
         defaultValue: false,
         description: "current day state",
         control: {
            type: 'boolean'
         }
      }
   }
}

const Template = (args) => CalendarButton(args);

export const Standard = Template.bind({});
