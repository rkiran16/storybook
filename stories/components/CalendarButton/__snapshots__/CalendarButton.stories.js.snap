// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Buttons/CalendarButton Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    ">
      ",
    "
      <input type=\\"checkbox\\" class=\\"btn-check\\" id=\\"btn-check\\" ?disabled=",
    " ?checked=",
    "  autocomplete=\\"off\\">
      <label class=\\"btn\\" for=\\"btn-check\\">
        <div class=\\"d-flex flex-column justify-content-evenly\\">
          <span class=\\"material-icons icon\\">done</span>
          <span class=\\"day\\">Sat</span>
          <span class=\\"date\\">5</span>
        </div>
      </label>
     </div>
   ",
  ],
  "values": Array [
    "calendar-btn-wrapper ",
    "",
    false,
    false,
  ],
}
`;
